### PR TITLE
have OMERO handle the new Folder model object

### PR DESCRIPTION
**Pull request by mtbc**
See https://github.com/openmicroscopy/bioformats/pull/2210. Introduces database patch `OMERO5.3DEV__1`. New integration tests for:

* http://regions-ci.docker.openmicroscopy.org:8080/job/OMERO-start/lastCompletedBuild/testReport/integration/PojosServiceTest/
* http://regions-ci.docker.openmicroscopy.org:8080/job/OMERO-start/lastCompletedBuild/testReport/integration/UpdateServiceTest/

Note availability of `P-TRE_12_R3D_D3D-folders.ome.tiff` sample file.

--no-rebase

### CODE REVIEW

**Comment by sbesson** [comment]
http://regions-ci.docker.openmicroscopy.org:8080/job/OMERO-start/2/: no failed test!

File: components/common/src/ome/api/IContainer.java [hunk 1]
```diff
@@ -32,8 +30,8 @@
  * The names of the methods correlate to how the function operates:
  * <ul>
  * <li><b>load</b>: start at container objects and work down toward the
- * leaves, returning hierarchy (Proejct-&gt;Dataset-&gt;Image</li>
- * <li><b>find</b>: start at leave objects and work up to containers,
+ * leaves, returning hierarchy (Project-&gt;Dataset-&gt;Image</li>
+ * <li><b>find</b>: start at leaf objects and work up to containers,
```
Comment (joshmoore) [inline components/common/src/ome/api/IContainer.java]: A heads up that with the move towards a merge workflow, this type of change should be avoided on the non-mainline perhaps even more than before.

Comment (mtbc) [inline components/common/src/ome/api/IContainer.java]: Agreed, I didn't originally realize how hard we were avoiding branching `dev_5_2`.

**Comment by dominikl** [comment]
Tried to build this branch, but get 7 errors of this kind:
```
/Users/dlindner/Repositories/openmicroscopy/components/blitz/src/ome/formats/OMEROMetadataStoreClient.java:4076: error: method does not override or implement a method from a supertype
    @Override
```

**Comment by mtbc** [comment]
OMERO must consume Bio-Formats merge develop JAR which includes https://github.com/openmicroscopy/bioformats/pull/2210.

**Comment by mtbc** [comment]
I should add: what I do is have the Bio-Formats merge branch checked out locally and do a `mvn install` there before building OMERO so that the JARs are already in my local maven cache.

**Comment by sbesson** [comment]
Coming back to @joshmoore's comment, thoughts about dropping d52cfb2ab281912679fa2327b24c27e955db6275 from this PR?

**Comment by mtbc** [comment]
Those regions of lines haven't been touched since early 2009 so I rather doubt they're about to conflict, and I'd expect that any changes to them are more likely to be made on this branch than on mainline, so I really doubt in this case it's a problem at all. Still, if you prefer, I can force-push it out and open a PR against develop instead ...

**Comment by sbesson** [comment]
Relaunching Travis with a clean cache now openmicroscopy/bioformats#2210 is merged into BIo-Formats `develop` and the latest JARs are deployed to Artifactory. On the previous comment, leaving @joshmoore and @chris-allan to comment on potential conflicts with `IContainer.java`. Otherwise all [integration tests](http://regions-ci.docker.openmicroscopy.org:8080/job/OMERO-start/18/) are passing with this PR included.

**Comment by sbesson** [comment]
Merging for now to simplify the rest of the model changes.

**Pull request merged**